src/Constraint/Contains.php

```diff
@@ -64,7 +64,7 @@ public function evaluate($value)
     public function equivalentTo(Expression $other)
     {
         // Since this class is final, we can check with instanceof
-        return $other instanceof $this && $this->comparedValue == $other->comparedValue;
+        return $other instanceof $this && (string) $this->comparedValue === (string) $other->comparedValue;
```

Comment:
you may even avoid the type casting IMO (or you should do it in the constructor)